pretyping/evarsolve.ml

```diff
@@ -1713,7 +1713,7 @@ and evar_define unify flags ?(choose=false) ?(imitate_defs=true) env evd pbty (e
     (* so we recheck acyclicity *)
     if occur_evar_upto_types evd' evk body then raise (OccurCheckIn (evd',body));
     (* needed only if an inferred type *)
-    let evd', body = refresh_universes pbty env evd' body in
+    let evd', body = refresh_universes ~status:UState.univ_flexible ~refreshset:true pbty env evd' body in
```

Comment:
I don't know if doing a rigid refresh was intentional here.

Reply:
This comes from 2005 (9061ea66e66) and I don't know either. Were there flexible universes at this time?

Reply:
There were no flexibles at the time, but we surely should use a flexible to refresh.